### ISSUE TITLE
Video tag does not seems to be parsed

### ISSUE THREAD

**Issue by AyWa** [otiai10/opengraph#20]
Video tag does not seems to be parsed. I will try to implement a base version of it with test

**Comment by otiai10** [comment]
@AyWa Can you give me some example pages including `video` tags?

**Comment by AyWa** [comment]
For sure, you can check any video on youtube, or any other website that might be display as video on twitter / FB etc.

https://www.youtube.com/watch?v=1MxA0i2rxQo

<img width="693" alt="Screen Shot 2020-12-11 at 2 00 01 PM" src="https://user-images.githubusercontent.com/13608477/101865287-32ab2e80-3bb9-11eb-95fb-21613f8d0fd2.png">


I plan to do simple implementation for it today :)

**Comment by otiai10** [comment]
Thx AyWa!
1 thing to add, I'm planning to release v2 in few weeks, can you send your PR to `develop`?
and any feedback on `develop` will be welcomed ;)

**Comment by AyWa** [comment]
Sure let me do that on `develop` branch :)
Should we include https://github.com/otiai10/opengraph/commit/7d11cc903f52dc66ec702cb21e45fd7b3d84ecdb to develop branch too ?

**Comment by otiai10** [comment]
I already rebased `develop` branch and `Walk` is there ;)
Can you please just branch out from the latest `develop` 🙏

**Comment by AyWa** [comment]
So I am testing `develop` branch, and it seems better than before 👍 

Just wonder if this part: 

https://github.com/otiai10/opengraph/blob/develop/opengraph.go#L127

can be inside the `opengraph.New()` ? because now I need to do something like:

```
		og := opengraph.New(reqURL)
		og.Intent.TrustedTags = []string{opengraph.HTMLMetaTag, opengraph.HTMLTitleTag, opengraph.HTMLLinkTag}
                og.Walk(....
```

**Comment by otiai10** [comment]
Thank you, good catch!

The only guy who needs to know what `TrustedTags` are is `walk`, therefore I'd prefer below:

```go
func (og *OpenGraph) walk(node *html.Node) error {

	if len(og.Intent.TrustedTags) == 0 {
		if og.Intent.Strict {
			og.Intent.TrustedTags = []string{HTMLMetaTag}
		} else {
			og.Intent.TrustedTags = []string{HTMLMetaTag, HTMLTitleTag, HTMLLinkTag}
		}
	}

	// following stuff here
}
```

Let me know your opinion.

**Comment by otiai10** [comment]
Wait, if it's like that, that code block will be executed for all `walk` which are called recursively.

Then I'd like to suggest:

```diff
func (og *OpenGraph) Parse(body io.Reader) error {
	// other stuff here
-	og.walk(node)
+	og.Walk(node)
	return nil
}

func (og *OpenGraph) Walk(n *html.Node) error {
+	if len(og.Intent.TrustedTags) == 0 {
+		if og.Intent.Strict {
+			og.Intent.TrustedTags = []string{HTMLMetaTag}
+		} else {
+			og.Intent.TrustedTags = []string{HTMLMetaTag, HTMLTitleTag, HTMLLinkTag}
+		}
+	}
	return og.walk(n)
}
```

**Comment by AyWa** [comment]
I think it is fine for me too 👍

**Comment by otiai10** [comment]
Closed by #21
